### PR TITLE
block player one from selecting a block with the wrong value

### PR DESCRIPTION
**Pull request by kwiedenhaupt**
used currentlySelectedBlock to store the index of the previously clicked block in order to be used later in the code.

### CODE REVIEW

File: pages/studentPortal/labs/multiplication/game.tsx [hunk 1]
```diff
@@ -67,79 +73,49 @@ export function showEndGameImage(array: GameBlockState[]) {
...
-      let initiateBlockState = {};
+  const dispatch = useDispatch();
+  const { stage, blocks: gameState } = useSelector(longestStreakSelector);
 
-      initiateBlockState = {
-        text: z.toString(),
-        state: BlockState.NOT_SELECTED,
-      };
-      dummyArray.push(initiateBlockState);
+  //const [stage, setStage] = useState(STAGE.SET_RULES);
```
Comment (vithushan19) [inline pages/studentPortal/labs/multiplication/game.tsx:79]: don't merge commented out code

File: redux/longestStreakSlice.ts [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+          if (state.blocks[index].value === state.blocks[firstSelectedBlockIndex].value) {
+            state.blocks[index].state = BlockState.PLAYER_ONE_SELECTED;
+            state.isPlayerSelecting = false;
+            console.log(index)
+            console.log(state.currentlySelectedBlock)
+            handleAISelection(state);
+            
+          }
+        
+        //if (state.blocks[index] === firstBlock.indexOf() ) {
```
Comment (vithushan19) [inline redux/longestStreakSlice.ts:119]: don't merge this in. please remove

**Review by vithushan19 — COMMENTED** [review]
You have too much commented out code, but everything else looks great. Please get Abhi to review your code and you can merge it once you get his approval

**Comment by vithushan19** [comment]
looks like your tests are also failing and need to be fixed

**Comment by vithushan19** [comment]
Your build is failing. Try `npm run build` to see the errors.

**Comment by vithushan19** [comment]
Let me know when your build is passing all the checks and then I'll review. You might want to ask abhi to review before me.

File: pages/studentPortal/labs/multiplication/game.tsx [hunk 3]
```diff
@@ -64,21 +63,21 @@ export function showEndGameImage(array: GameBlockState[]) {
...
 export default function BlockComponentGallery() {
   const dispatch = useDispatch();
-  const { stage, blocks: gameState } = useSelector(longestStreakSelector);
-
-  //const [stage, setStage] = useState(STAGE.SET_RULES);
-
-  const [playerOneName, setPlayerOneName] = useState("Player 1");
+  const {
+    stage,
+    blocks: gameState,
```
Comment (vithushan19) [inline pages/studentPortal/labs/multiplication/game.tsx:78]: Do you understand what's happening here? This is called json destructing

File: pages/studentPortal/labs/multiplication/game.tsx [hunk 4]
```diff
@@ -139,16 +140,16 @@ export default function BlockComponentGallery() {
                 ))
                 .reverse()}
             </div>
-            <div className="col-span-7 bg-rattata">
+            <div className="col-span-7 bg-gradient-to-r from-rattata ...">
```
Comment (vithushan19) [inline pages/studentPortal/labs/multiplication/game.tsx:143]: ... ?

File: pages/studentPortal/labs/multiplication/game.tsx [hunk 5]
```diff
@@ -15,10 +16,10 @@ import {
   setStage,
   STAGE,
 } from "../../../../redux/longestStreakSlice";
-
 export type GameBlockState = {
   text: string;
   value: number;
+  isProduct?: boolean;
```
Comment (vithushan19) [inline pages/studentPortal/labs/multiplication/game.tsx:22]: Why is this needed?

File: redux/longestStreakSlice.ts [hunk 6]
```diff
@@ -22,36 +24,50 @@ export enum STAGE {
...
+  blocks: initializeGameState(),
+  handlePlayerSelect: 0,
+  isPlayerSelecting: false,
+  currentlySelectedBlock: null,
+  playerName: "Player 1",
+};
+
+const resetInitialState: LongestStreakState = {
+  stage: STAGE.PLAY_GAME,
+  reset: false,
```
Comment (vithushan19) [inline redux/longestStreakSlice.ts:65]: Why is this property needed?

Comment (vithushan19) [reply redux/longestStreakSlice.ts:65]: When is it used? When is it set to true?

File: redux/longestStreakSlice.ts [hunk 7]
```diff
@@ -60,94 +76,145 @@ export const longestStreakSlice: Slice = createSlice({
...
-        if (state.isPlayerSelecting === false && unselectedBlocks.includes(state.blocks[index])) {
-         // Player selected first block
-          state.blocks[index].state = BlockState.PLAYER_ONE_SELECTED;
+        const secondSelectedBlockIndex = state.currentlySelectedBlock;
+        // only do this if player clicked on a non selected block
+        if (
+          state.isPlayerSelecting === false &&
+          unselectedBlocks.includes(state.blocks[index])
+        ) {
+          state.blocks[index].state = BlockState.HIGHLIGHTED;
```
Comment (vithushan19) [inline redux/longestStreakSlice.ts:132]: Cool!

File: pages/studentPortal/labs/multiplication/game.tsx (hunk 5 above)
Comment (vithushan19) [reply pages/studentPortal/labs/multiplication/game.tsx:22]: If it's for ai selection why did you make this property an optional?

**Review by vithushan19 — APPROVED** [review]
Nice! I had a few questions but it looks good to me.

I think we are almost done with this game. Next week you will learn about databases and how to read and write to a db. You will add a concept of a user level for this game. When they beat the first level they progress to their 3 times tables.

When your game is complete we will do a test session with the other students where they will look for bugs in your game. So make sure you are testing different edge cases and you're ready for others to try to break your game.

Great work!

File: pages/studentPortal/labs/multiplication/game.tsx (hunk 3 above)
Comment (kwiedenhaupt) [reply pages/studentPortal/labs/multiplication/game.tsx:78]: I understand that it's tying the component to the state, doing the getting and the setting and tying it to the payload action.  That's just a more succinct way to do it?

File: pages/studentPortal/labs/multiplication/game.tsx (hunk 4 above)
Comment (kwiedenhaupt) [reply pages/studentPortal/labs/multiplication/game.tsx:143]: Screen Shot 2022-09-16 at 7 38 03 PM
adds a fade to transparent gradient to the center of the game board

File: redux/longestStreakSlice.ts (hunk 6 above)
Comment (kwiedenhaupt) [reply redux/longestStreakSlice.ts:65]: It allows me to re-initialize the game board with redux when I hit the reset button on the play_game stage. It gives me new numbers, etc.

Comment (kwiedenhaupt) [reply redux/longestStreakSlice.ts:65]: I had to clear state for the game in order to truly reset. Otherwise, if I just tried to initialize it would remember some parts of the state.
